src/nodes/statements.ts

```diff
@@ -71,16 +71,30 @@ export class CEmptyStatement {
 }
 
 @CodeTemplate(`
-{destructors}
-return {expression};
+{
+    {returnTypeAndVar} = {expression};
+    {destructors}
+    return {returnTemp};
+}
```

Comment:
Would be great if we could avoid creating a variable in every function where we want to return smth.
Consider e.g. smth like
```
function sum(a, b) { return a + b; }
console.log(sum(10,20));
```
Do I understand correctly, that with this code it will create a completely unnecessary `__fake` variable?

So instead of
```
int sum(int16_t a, int16_t b) {
    return a + b;
}
int main() {
    printf(sum(10, 20));
}
```

it will be
```
int sum(int16_t a, int16_t b) {
    int16_t __fake;
    {
        __fake = a + b;
        return __fake;
    }
}
int main() {
    printf(sum(10, 20));
}
```

TS2C should not produce anything that is not needed

Reply:
no? that isn't how it works. A) the "fake" thing isn't actually a variable, it is a hack to get array passing to work. Additionally, this generates the exact same code, byte for byte:
```
sum:
 lea    eax,[rsi+rdi*1]
 ret
sum2:
 lea    eax,[rsi+rdi*1]
 ret
```
That is because GCC ends up putting it into EAX anyway, so there is no difference in the actual assembly. Also, the variable is called `returnTemp`. I know my code is confusing. This variable is actually needed:
```c
// here we can safely load returnTemp:
short returnTemp = obj->x;
// destructors run here, we need to get the result before this.
free(obj);
return returnTemp;
```
With my understanding of C, this is necessary to do.

Reply:
But yes, this does create an extra variable every time.

Reply:
Fixed